src/main/java/frc/robot/subsystems/BallPath/Shooter/ShooterImpl.java

```diff
@@ -38,11 +103,253 @@ public int checkBalls(){
     }
 
     @Override
-    public void setHoodAngle(double distance){}
+    public void setHoodAngle(double setPointHood){
+        if(turretHoodPosition >= setPointHood - hoodBuffer && turretHoodPosition <= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, 0);
+          hoodReady = false;
+        }else if(turretHoodPosition <= setPointHood - hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, hoodSpeed);
+          hoodReady = false;
+        }else if (turretHoodPosition >= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, -hoodSpeed);
+          hoodReady = false;
+        }
+    }
 
     @Override
     public void start(){}
 
     @Override
-    public void stop(){}
+    public void stop(){
+        System.out.println("DISABLING THE SHOOTER");
+        setPointHood = 0;
+        setPointRotation = 0;
+
+        shooterMotor.set(ControlMode.PercentOutput, 0);
+
+        if(turretEncoderReadingPosition >= setPointRotation - turretBuffer && turretEncoderReadingPosition <= setPointRotation + turretBuffer){
+          turretMotor.set(ControlMode.PercentOutput, 0);
+          turretReady = false;
+        }else if(turretEncoderReadingPosition <= setPointRotation - turretBuffer){
+          turretMotor.set(ControlMode.PercentOutput, turretSpeed);
+          turretReady = false;
+        }else if (turretEncoderReadingPosition >= setPointRotation + turretBuffer){
+          turretMotor.set(ControlMode.PercentOutput, -turretSpeed);
+          turretReady = false;
+        }
+        
+        if(turretHoodPosition >= setPointHood - hoodBuffer && turretHoodPosition <= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, 0);
+          hoodReady = true;
+        }else if(turretHoodPosition <= setPointHood - hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, hoodSpeed);
+          hoodReady = false;
+        }else if (turretHoodPosition >= setPointHood + hoodBuffer){
+          hoodMotor.set(ControlMode.PercentOutput, -hoodSpeed);
+          hoodReady = false;
+        }
+
+    }
+
+    @Override
+    public void shootFender() {
+        // Fender shot for teleop
+        System.out.println("FENDER SHOT: ");
+        setPointHood = 100000;
+        // setPointShooter = 0.33;
+        setPointShooterPID = 0; // tbd
+        setPointRotation = 0;
+        System.out.println("SETPOINT HOOD: " + setPointHood);
+        System.out.println("SETPOINT SHOOTER " + setPointShooter);
+        System.out.println("SETPOINT FOR THE TURRET TURNING " + setPointRotation);
+
+        // if we dont want to use PID, use the below
+        // shooterMotor.set(ControlMode.PercentOutput, setPointShooter);
+        // System.out.println(setPointShooter);
+
+        // pid for shooter
+        if(setPointShooter != 0){
```

Comment:
This is referencing `setPointShooter`, despite that it's using PID, and there is also a `setPointShooterPID` field. Seems confused and confusing. Right now these methods will never run the shooter because this wrong variable is being referenced. Could you remove the `setPointShooter` variable and old commented out code that was using raw motor percentages? If we want to keep that around as a fallback we can add that as a separate Shooter implementation and we can easily switch between them as required by changing one initialization in `Robot.java`, rather than having to comment and un-comment large portions of the same file to switch between implementations.